### PR TITLE
feat: Thumbnailing - backend implementation

### PR DESCRIPTION
**Pull request by loggerz**
Finally a proof of concept for the thumbnailling using MuPDF.js : a pure javascript wrapper providing a webassembly implementation of MuPDF.

Todo:
- [x] Quick POC of PDF to image conversion
- [x] Refactor of previous code to distinguish the specific parts and the common ones
- [x] Proper implementation of PDF to image conversion
- [x] Quick POC of image downscaling
- [x] Proper implementation of image downscaling

### CODE REVIEW

File: cloud-functions/functions/src/thumbnailing.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * Convert a pdf to a png image
+ * @param {ArrayBuffer} buffer The pdf to convert
+ * @return {Uint8Array} The png image
+ */
+function convertPdfToImage(buffer: ArrayBuffer): ArrayBuffer {
+  const originalDocument = mupdf.Document.openDocument(buffer, "application/pdf");
+  const page = originalDocument.loadPage(0);
+  const matrix = mupdf.Matrix.scale(PDF_CONVERSION_DPI / 72, PDF_CONVERSION_DPI / 72);
```
Comment (ArthurChalard) [inline cloud-functions/functions/src/thumbnailing.ts:31]: What is this 72 ? It could be interesting to put it in a named constant

**Review by ArthurChalard — CHANGES_REQUESTED** [review]
You made great refactoring of the existing cloud function to separate the code in multiple files. However, you could have done this in an other PR to simplify the reviewing process.
The new code is well modularized and you log regularly your state so that the debug process will be easier if we encounter a failure.

File: cloud-functions/functions/src/thumbnailing.ts (hunk 1 above)
Comment (loggerz) [reply cloud-functions/functions/src/thumbnailing.ts:31]: Indeed, the 72 is the default DPI for PDFs in the context of converting it to an image. The other constant is the expected DPI for the output. I will rename them

**Comment by loggerz** [comment]
I agree with you for the refactoring, it didn't help the review. The reason is that I started implementing on the single file but it grew to almost a thousand lines hence the included refactoring

**Review by ArthurChalard — APPROVED** [review]
The modifications I asked are made. The rest looks good to me.

**Pull request merged**